### ISSUE TITLE
Implement function to create shapefile of connections in MgmtData

### ISSUE THREAD

**Issue by jcmand** [rcapell/RHYPE#26]
Implement function to create shapefile & plot preview of regsrcid connections in MgmtData.txt

**Comment by rcapell** [comment]
Jafet, could this be implemented without importing the maptools library? 
RHYPE currently imports sp. 
I don't know which function(s) is/are maptools, but as a general rule, unnecessary dependencies should be avoided and maybe there are comparable functions in sp?

**Comment by jcmand** [comment]
I use maptools to read a point shapefile and to write a line shapefile. I have not found these functions in sp, but perhaps I missed this. I am happy to change if you have an alternative.

**Comment by rcapell** [comment]
Ok, I see. So far I have not included import/export from/to certain formats
in computational functions. I will do the core function of your script
first (for which sp suffices), and then you can evaluate.

**Comment by dgustafsson** [comment]
I am using the readOGR and writeOGR functions in the rgdal library. The advantage of these functions is taht they also read projection information, and it is then easy to use the spTransform function for further transformations,

**Comment by dgustafsson** [comment]
sorry for closing

**Comment by rcapell** [comment]
This is now implemented as MapRegionalSources(), please test if it works as expected. The function covers step 3 in the script above.

I replaced the for loop in your script with a lapply()/sapply() combination, and have the -unverified- impression that it is faster this way. 

@jcmand Please close this issue if you are satisfied with the solution.

Example for how to use it
```
library(rgdal)
mydata <- ReadMgmtData(filename="MgmtData.txt")
mymap <- readOGR(dsn = "d:/Current_shapefiles", layer = "SUBID_CenterPoints")
te <- MapRegionalSources(data = mydata, map = mymap)
plot(te)
writeOGR(obj = te, dsn = "d:/", layer = "SUBID_REGSRCID_lines", driver = "ESRI Shapefile", verbose = T)
```

**Comment by rcapell** [comment]
closing for now.

**Comment by jcmand** [comment]
Hej Rene!
Sorry att jag inte kollat på detta! Det har varit för mkt annat. Det är säkert bra. Jag får kolla framgent.
/Jafet

Från: rcapell [mailto:notifications@github.com]
Skickat: 19 May 2015 15:33
Till: rcapell/RHYPE
Kopia: Andersson Jafet
Ämne: Re: [RHYPE] Implement function to create shapefile of connections in MgmtData (#26)


closing for now.

**Comment by jcmand** [comment]
I have now tested that the MapRegionalSources() works as expected. It produces the same output as my previous script for the EHYPE 3.0 setup. I checked the number of links, the link-2-link connections, as well as the distances. For the latter I had a rounding to 3 decimals which is not in the MapRegionalSources(), but that is a minor detail and a user can easily round the data if desired. 
The apply() solution is most probably much faster and preferred.
/Jafet
